doc/user/content/sql/create-source/_index.md

```diff
@@ -20,6 +20,31 @@ Sources represent connections to resources outside Materialize that it can read
 data from. For more information, see [API Components:
 Sources](../../overview/api-components#sources).
 
+## Materialized source details
+
+Materializing a source keeps data it receives in an in-memory
+[index](/overview/api-components/#indexes), the presence of which makes the
+source directly queryable. In contrast, non-materialized sources cannot process
+queries directly; to access the data the source receives, you need to create
+[materialized views](/sql/create-materialized-view) that `SELECT` from the
+source.
```

Comment:
```suggestion
Materializing a source keeps data it receives in an in-memory
[index](/overview/api-components/#indexes), which makes the
source directly queryable. In contrast, non-materialized sources cannot process
queries directly; to access the data the source receives, you need to create
[materialized views](/sql/create-materialized-view) that `SELECT` from the
source.
```

Reply:
Nit, but this bit has always struck me as unnecessarily formal.

Reply:
my thoughts, not strongly held:

I like Marta's idea of adding a brief summary of `MATERIALIZED` on the `API Components` page, and adding a much more detailed summary as-is.  If we do that, I'm also a fan of surfacing the `MATERIALIZED` concept out and making it more visible, as it's a critical part of what we do, and will impact the user's experience with the product.

However, I can also see the value of refactoring all the redundant source info to it's own, dedicated SOURCE page!  Maybe if we choose to proceed with the refactor we decide to add more of the common, cross-source items, so it doesn't confuse users which page they should go to?

Reply:
I've followed Marta's suggestion for location, leaving the content largely as-is. I like the idea of creating a separate page for materialization in general, but I want to think about it more before committing.